conda/core/prefix_data.py

```diff
@@ -84,6 +100,35 @@ def __init__(
             else context.pip_interop_enabled
         )
 
+    @classmethod
+    def from_name(cls, name: str, **kwargs):
```

Comment:
If we expect these to be used from outside of this project (which they should!), we need to make sure to add a nice doc string to the method clearly stating why this method should be used and what it returns.

Reply:
Yep, didn't put the time on these yet because I didn't know how many of these new API endpoints would survive a review :P Looks like we are keeping them so I'll write them down.